### PR TITLE
Allows filling welders from open containers that have welding fuel in them

### PR DESCRIPTION
**Pull request by PKPenguin321**
🆑
tweak: You can now use beakers/cups/etc that have welding fuel in them on welders to refuel them.
/:cl:

Closes https://github.com/tgstation/tgstation/issues/31339

### CODE REVIEW

**Comment by kevinz000** [comment]
@PKPenguin321 no message for the user..?

**Comment by PKPenguin321** [comment]
I knew I was forgetting something

File: code/game/objects/items/tools.dm [hunk 1]
```diff
@@ -457,6 +457,12 @@
 		flamethrower_screwdriver(I, user)
 	else if(istype(I, /obj/item/stack/rods))
 		flamethrower_rods(I, user)
+	else if(istype(I, /obj/item/reagent_containers) && (I.container_type & OPENCONTAINER_1))
+		var/amountNeeded = max_fuel - get_fuel()
+		var/obj/item/reagent_containers/container = I
+		if(amountNeeded > 0 && container.reagents.has_reagent("welding_fuel"))
+			container.reagents.trans_id_to(src, "welding_fuel", amountNeeded)
+			to_chat(user, "<span class='notice'>You transfer some fuel from \the [container] to \the [src].</span>")
```
Comment (PKPenguin321) [inline code/game/objects/items/tools.dm]: i always forget if `\the` is the correct way to do this please correct me if im wrong

**Comment by RandomMarine** [comment]
Rigging welders to explode by filling them with plasma **_WHEN?_**

**Comment by PKPenguin321** [comment]
It would actually not be hard

**Comment by PKPenguin321** [comment]
I will PR it after this goes through, since doing so beforehand would cause conflicts and putting it in the same PR pushes this too far into balance change territory

File: code/game/objects/items/tools.dm (hunk 1 above)
Comment (JJRcop) [reply code/game/objects/items/tools.dm]: Nowadays we just do ```[container]``` and ```[src]```. ```\the``` does work still, but we tend on not using it.

Comment (PKPenguin321) [reply code/game/objects/items/tools.dm]: So it's fine to leave as is then

**Comment by ma44** [comment]
>fixing a 'not a bug'
:thinking:

File: code/game/objects/items/tools.dm (hunk 1 above)
Comment (ShizCalev) [reply code/game/objects/items/tools.dm]: Remove \the

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
Spessmen have some very steady hands and unbreakable patience if they can separate 1 unit of welding fuel from 99 units of water and transfer it

**Comment by PKPenguin321** [comment]
>Spessmen have some very steady hands and unbreakable patience if they can separate 1 unit of welding fuel from 99 units of water and transfer it

fixed

**Comment by PKPenguin321** [comment]
no! my gbp!!!

**Comment by PKPenguin321** [comment]
for real though i think this is good for merge now

File: code/game/objects/items/tools.dm [hunk 2]
```diff
@@ -457,6 +457,15 @@
 		flamethrower_screwdriver(I, user)
 	else if(istype(I, /obj/item/stack/rods))
 		flamethrower_rods(I, user)
+	else if(istype(I, /obj/item/reagent_containers) && is_open_container())
```
Comment (ShizCalev) [inline code/game/objects/items/tools.dm]: @optimumtact Shouldn't this have been I.is_open_container()???????

Comment (PKPenguin321) [reply code/game/objects/items/tools.dm]: uhhh

Comment (PKPenguin321) [reply code/game/objects/items/tools.dm]: yes

Comment (PKPenguin321) [reply code/game/objects/items/tools.dm]: to be fair all this would do is allow you to use pills to fuel it

**Pull request merged**